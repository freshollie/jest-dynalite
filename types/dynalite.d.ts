declare module "dynalite" {
  import { Server } from "http";

  interface DynaliteOptions {
    ssl?: boolean;
    path?: string;
    createTableMs?: number;
    deleteTableMs?: number;
    updateTableMs?: number;
    maxItemSizeKb?: number;
  }

  function createDynalite(
    options: DynaliteOptions
  ): Server & { close(cb?: (err?: Error) => void | undefined): void };

  export default createDynalite;
}
