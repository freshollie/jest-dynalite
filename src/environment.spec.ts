import { join } from "path";
import { Config } from "@jest/types";
import Environment from "./environment";
import { dynaliteInstance } from "./db";

jest.mock("dynalite", () => () => {
  let listening = false;
  return {
    get listening() {
      return listening;
    },
    listen: (_: void, resolve: () => void) => {
      listening = true;
      resolve();
    },
    close: (resolve: () => void) => {
      listening = false;
      resolve();
    },
  };
});

describe("Environment", () => {
  it("should throw an error if config file could not be located", () => {
    expect(
      () =>
        new Environment({
          rootDir: "somebaddirectory",
          testEnvironmentOptions: {},
        } as Config.ProjectConfig)
    ).toThrowErrorMatchingSnapshot();
  });

  it("should throw a different error if a jestDynaliteConfigDirectory is passed to the test environment", () => {
    expect(
      () =>
        new Environment({
          testEnvironmentOptions: {
            jestDynaliteConfigDirectory: "somebaddirectory",
          } as Record<string, unknown>,
        } as Config.ProjectConfig)
    ).toThrowErrorMatchingSnapshot();
  });

  it("should not throw an error if a valid config directory is given", () => {
    expect(
      () =>
        new Environment({
          rootDir: join(__dirname, "__testdir__"),
          testEnvironmentOptions: {},
        } as Config.ProjectConfig)
    ).not.toThrowError();
  });

  it("should start the database when 'setup' is called and stop the db when 'teardown' is called", async () => {
    const environment = new Environment({
      rootDir: join(__dirname, "__testdir__"),
      testEnvironmentOptions: {},
    } as Config.ProjectConfig);
    await environment.setup();
    expect(dynaliteInstance.listening).toBeTruthy();
    await environment.teardown();
    expect(dynaliteInstance.listening).toBeFalsy();
  });

  it("should should work when config directory is specified", async () => {
    const environment = new Environment({
      testEnvironmentOptions: {
        jestDynaliteConfigDirectory: join(__dirname, "__testdir__"),
      } as Record<string, unknown>,
    } as Config.ProjectConfig);
    await environment.setup();
    expect(dynaliteInstance.listening).toBeTruthy();
    await environment.teardown();
    expect(dynaliteInstance.listening).toBeFalsy();
  });
});
