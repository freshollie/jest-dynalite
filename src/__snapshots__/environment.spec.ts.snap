// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Environment should throw a different error if a jestDynaliteConfigDirectory is passed to the test environment 1`] = `
"
jest-dynalite could not find \\"jest-dynalite-config.js\\" or \\"jest-dynalite-config.ts\\" in the provided jestDynaliteConfigDirectory (somebaddirectory).

If you didn't intend to be using this directory for the config, please specify a custom
directory: https://github.com/freshollie/jest-dynalite/#advanced-setup

If you are not using the preset, you should apply 'import \\"jest-dynalite/withDb\\"'
to your \\"setupFilesAfterEnv\\".

For more information, please see https://github.com/freshollie/jest-dynalite/#breaking-changes.
      "
`;

exports[`Environment should throw an error if config file could not be located 1`] = `
"
jest-dynalite could not find \\"jest-dynalite-config.js\\" or \\"jest-dynalite-config.ts\\" in the jest <rootDir> (somebaddirectory).

If you didn't intend to be using this directory for the config, please specify a custom
directory: https://github.com/freshollie/jest-dynalite/#advanced-setup

If you are not using the preset, you should apply 'import \\"jest-dynalite/withDb\\"'
to your \\"setupFilesAfterEnv\\".

For more information, please see https://github.com/freshollie/jest-dynalite/#breaking-changes.
      "
`;
